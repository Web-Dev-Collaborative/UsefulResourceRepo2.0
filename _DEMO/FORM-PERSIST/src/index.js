import React from "react";
import { render } from "react-dom";
import { createPersistDecorator } from "final-form-persist";
import { Form, Field } from "react-final-form";

const App = () => {
  const { persistDecorator, clear } = createPersistDecorator({
    name: "myPersistKey",
    debounceTime: 500,
    whitelist: ["name"],
  });
  const initialValues = {
    name: "Hello",
  };

  return (
    <Form
      decorators={[persistDecorator]}
      initialValues={initialValues}
      onSubmit={() => console.log("on submit")}
    >
      {({ form: { reset } }) => (
        <form>
          Name:
          <Field name="name">
            {({ input }) => <input placeholder="Name" {...input} />}
          </Field>
          Password
          <Field name="password">
            {({ input }) => <input placeholder="Password" {...input} />}
          </Field>
          <button
            type="button"
            onClick={() => {
              reset(initialValues);
              clear();
            }}
          >
            Reset
          </button>
        </form>
      )}
    </Form>
  );
};

const rootElement = document.getElementById("root");
render(<App />, rootElement);
