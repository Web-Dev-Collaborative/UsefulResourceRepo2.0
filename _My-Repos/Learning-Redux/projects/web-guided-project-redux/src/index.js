/*
Our Reducer:

export const initialState = {
    appName: "Redux Test",
    title: "Dragon Member List 🐲",
    editing: false
  };

const titleReducer = (state, action) => {
  switch (action.type) {
    case "UPDATE_TITLE":
      return {
        ...state,
        title: action.payload,
        editing: false
      };
    case "TOGGLE_EDITING":
      return {
        ...state,
        editing: !state.editing
      };
    default:
      return state;
  }
};

export default titleReducer;

*/
import React from "react";
import ReactDOM from "react-dom";

import { createStore } from "redux";
import { Provider } from "react-redux";

import reducer from "./reducers/titleReducer";

import Title from "./components/Title";
import DragonList from "./components/DragonList";
import "./styles.css";

const store = createStore(reducer);

function App() {
  return (
    <div className="App">
      <Title />
      <DragonList />
    </div>
  );
}

const rootElement = document.getElementById("root");
ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  rootElement
);
