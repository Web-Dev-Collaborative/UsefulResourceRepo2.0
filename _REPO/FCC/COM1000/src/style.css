body {
  font-family: 'Ubuntu';
}

.menu > div >div:nth-child(2) {
  bottom: 0px;
  height: auto!important;
}

.ReactModalPortal * {
  z-index: 10;
}

.selectors {
  display: flex;
  width: 100%;
}

.selectors > .tabSelector {
  flex: 1 0 auto;
  background-color: rgb(51, 51, 51);
  color: white;
  text-align: center;
  margin: 4px;
  padding: 4px;
}