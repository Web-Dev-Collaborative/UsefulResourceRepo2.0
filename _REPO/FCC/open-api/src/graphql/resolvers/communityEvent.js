import {
  getCommunityEvent,
  getCommunityEvents,
  createCommunityEvent,
  deleteCommunityEvent
} from '../../dataLayer/mongo/communityEvent';

export const communityEventResolvers = {
  Query: {
    getCommunityEvent,
    getCommunityEvents
  },
  Mutation: {
    createCommunityEvent,
    deleteCommunityEvent
  }
};
