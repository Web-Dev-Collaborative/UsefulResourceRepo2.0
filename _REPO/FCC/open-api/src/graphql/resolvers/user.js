import { createUser, getUser, deleteUser } from '../../dataLayer/mongo/user';

export const userResolvers = {
  Query: {
    getUser
  },
  Mutation: {
    createUser,
    deleteUser
  }
};
