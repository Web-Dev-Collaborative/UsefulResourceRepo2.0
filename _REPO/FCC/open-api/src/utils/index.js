export { default as asyncErrorHandler } from './asyncErrorHandler';
