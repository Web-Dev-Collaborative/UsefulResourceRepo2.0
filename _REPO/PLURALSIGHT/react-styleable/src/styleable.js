import invariant from 'invariant'
import PropTypes from 'prop-types'
import React from 'react'

import getDisplayName from './utils/get-display-name'

const isSpecialKey = selector => selector === 'compose'

function getSelectorsInSetDifference(newCss = {}, origCss = {}) {
  const newSet = Object.keys(newCss)
  const origSet = Object.keys(origCss)
  return newSet
    .filter(selector => origSet.indexOf(selector) === -1)
    .filter(selector => !isSpecialKey(selector))
}

function areSelectorsMatchingSet(newCss, origCss) {
  return getSelectorsInSetDifference(newCss, origCss).length <= 0
}

const rmSpecialCss = css => {
  const { compose, ...rest } = css
  return rest
}

const compose = (toCompose = {}, css) =>
  Object.keys(toCompose).reduce((acc, selector) => {
    acc[selector] = css[selector] + ' ' + toCompose[selector]
    return acc
  }, css)

const validate = (origCss, newCss) => {
  invariant(
    areSelectorsMatchingSet(newCss, origCss),
    'Expected "this.props.css" to provide only selectors in the original stylesheet.  These selectors "%s" are not included in the stylesheet keys, "%s".',
    getSelectorsInSetDifference(newCss, origCss),
    Object.keys(origCss)
  )

  invariant(
    areSelectorsMatchingSet(newCss.compose, origCss),
    'Expected "this.props.css.compose" to provide only selectors in the original stylesheet.  These selectors "%s" are not included in the stylesheet keys, "%s".',
    getSelectorsInSetDifference(newCss.compose, origCss),
    Object.keys(origCss)
  )
}

export default function styleable(origCss) {
  if (!origCss) origCss = {}

  if (typeof origCss !== 'object' || Array.isArray(origCss))
    throw new Error(
      'stylesheet must be an object (ie, export object from a css module)'
    )

  return function decorateSource(DecoratedComponent) {
    class Styleable extends React.Component {
      getCss() {
        const newCss = this.props.css || {}
        validate(origCss, newCss)
        const overridden = { ...origCss, ...rmSpecialCss(newCss) }
        const composed = compose(newCss.compose, overridden)
        return composed
      }
      render() {
        return <DecoratedComponent {...this.props} css={this.getCss()} />
      }
    }
    Styleable.displayName = `Styleable(${getDisplayName(DecoratedComponent)})`
    Styleable.defaultProps = {
      ...DecoratedComponent.defaultProps,
      css: {}
    }
    Styleable.propTypes = {
      ...DecoratedComponent.propTypes,
      css: PropTypes.object
    }
    return Styleable
  }
}
