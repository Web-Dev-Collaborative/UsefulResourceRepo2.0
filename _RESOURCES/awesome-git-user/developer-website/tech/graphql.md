---
id: 4
title: GraphQL
tag: graphql
date: "2018-10-17"
image: ../src/images/icons/graphql.svg
pagetype: tech
---
