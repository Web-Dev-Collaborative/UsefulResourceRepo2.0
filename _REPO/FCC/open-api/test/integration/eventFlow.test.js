/* global beforeAll afterAll expect */
import mongoose from 'mongoose';
import { graphql } from 'graphql';
import { createUser } from '../../src/dataLayer/mongo/user';
import { createCommunityEvent } from '../../src/dataLayer/mongo/communityEvent';
import { graphqlSchema } from '../../src/handler';

const contextNoToken = global.mockedContextWithOutToken;
const invalidContext = global.mockedContextWithInValidToken;
const validContextCharlie = global.mockedContextWithValidTokenForCharlie;
const contextNoEmail = global.mockedContextWithNoEmailToken;
const validContextForBrian = global.mockedContextWithValidTokenForBrian;
const validContextForDennis = global.mockedContextWithValidTokenForDennis;

const event = {
  title: 'epoch',
  description: 'The start of POSIX time',
  date: 'Thu 1 Jan 1970 00:00:00'
};

beforeAll(async function beforeAllTests() {
  await mongoose.connect(global.__MONGO_URI__);

  // Create two test users
  await createUser({}, {}, validContextForBrian);
  await createUser({}, {}, validContextForDennis);

  // Create a test event
  await createCommunityEvent({}, event, validContextForBrian);
});

afterAll(async function afterAllTests() {
  await mongoose.connection.db.dropDatabase();
  await mongoose.disconnect();
});

// language=GraphQL

const createCommunityEventQuery = `
mutation createCommunityEvent {
  createCommunityEvent(
    title: "epoch"
    description: "The start of POSIX time"
    date: "Thu 1 Jan 1970 00:00:0"
    attendees: [{email: "dennisritchie@example.com"}]
  ) {
    title
    description
    owner {
      email
      externalId
    }
  }
}
`;

const expectedCommunityEventQuery = `
    query {
        getCommunityEvent(title: "epoch") {
          title
          description
          owner {
            email
          }
        }
    }
  `;

const expectedNoCommunityEventQuery = `
    query {
      getCommunityEvent(title: "yeah nah") {
        title
        description
        owner {
          email
        }
      }
    }
  `;

const skippedMandatoryFieldQuery = `
  query {
    getCommunityEvent {
      title
      description
      owner {
        email
      }
    }
  }
`;

const malformedQuery = `
    query {
      getCommunityEvent(externalId: "yeah nah") {
        title
        description
        owner {
          email
        }
      }
    }
  `;

const rootValue = {};

describe('createEvent', () => {
  it('should return null and an auth error without a token', done => {
    expect.assertions(1);

    graphql(graphqlSchema, createCommunityEventQuery, rootValue, contextNoToken)
      .then(({ errors }) => {
        expect(errors).toMatchSnapshot('not logged in');
        return;
      })
      .then(done)
      .catch(done);
  });

  it('should return null and an auth error with an invalid token', done => {
    expect.assertions(1);

    graphql(graphqlSchema, createCommunityEventQuery, rootValue, invalidContext)
      .then(({ errors }) => {
        expect(errors).toMatchSnapshot('no auth error');
        return;
      })
      .then(done)
      .catch(done);
  });

  it('should create an event by query', done => {
    expect.assertions(2);

    graphql(
      graphqlSchema,
      createCommunityEventQuery,
      rootValue,
      validContextForBrian
    )
      .then(({ data, errors }) => {
        expect(data.createCommunityEvent).toMatchSnapshot('new event');
        expect(errors).toMatchSnapshot('no errors');
        return;
      })
      .then(done)
      .catch(done);
  });

  it('should raise an error without email in token', done => {
    expect.assertions(2);

    graphql(graphqlSchema, createCommunityEventQuery, rootValue, contextNoEmail)
      .then(({ data, errors }) => {
        expect(data.createCommunityEvent).toMatchSnapshot('null');
        expect(errors).toMatchSnapshot('no email');
        return;
      })
      .then(done)
      .catch(done);
  });
});

describe('getEvent', () => {
  it('should return an event after one has been created', done => {
    expect.assertions(2);

    graphql(
      graphqlSchema,
      expectedCommunityEventQuery,
      rootValue,
      validContextCharlie
    )
      .then(({ data, errors }) => {
        expect(data.getCommunityEvent).toMatchSnapshot('event found');
        expect(errors).toMatchSnapshot('no errors');
        return;
      })
      .then(done)
      .catch(done);
  });

  it('should return null if no event has been found', done => {
    expect.assertions(2);

    graphql(
      graphqlSchema,
      expectedNoCommunityEventQuery,
      rootValue,
      validContextCharlie
    )
      .then(result => {
        const { data, errors } = result;
        expect(data.getCommunityEvent).toMatchSnapshot('null event');
        expect(errors).toMatchSnapshot('no event error');
        return;
      })
      .then(done)
      .catch(done);
  });

  it('should return errors for a query skipping a mandatory field', done => {
    expect.assertions(1);

    graphql(
      graphqlSchema,
      skippedMandatoryFieldQuery,
      rootValue,
      validContextCharlie
    )
      .then(result => {
        const { errors } = result;
        expect(errors).toMatchSnapshot('skipped mandatory field error');
        return;
      })
      .then(done)
      .catch(done);
  });

  it('should return errors for a malformed query', done => {
    expect.assertions(2);

    graphql(graphqlSchema, malformedQuery, rootValue, validContextCharlie)
      .then(result => {
        const { data, errors } = result;
        expect(data.getCommunityEvent).toBe(null);
        expect(errors).toMatchSnapshot('malformed query error');
        return;
      })
      .then(done)
      .catch(done);
  });
});
