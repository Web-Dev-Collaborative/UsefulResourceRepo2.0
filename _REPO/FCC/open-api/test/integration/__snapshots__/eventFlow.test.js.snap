// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createEvent should create an event by query: new event 1`] = `
Object {
  "description": "The start of POSIX time",
  "owner": Object {
    "email": "briankernighan@example.com",
    "externalId": null,
  },
  "title": "epoch",
}
`;

exports[`createEvent should create an event by query: no errors 1`] = `undefined`;

exports[`createEvent should raise an error without email in token: no email 1`] = `
Array [
  [GraphQLError: You must provide a valid email],
]
`;

exports[`createEvent should raise an error without email in token: null 1`] = `null`;

exports[`createEvent should return null and an auth error with an invalid token: no auth error 1`] = `
Array [
  [GraphQLError: You are not authorized, jwt malformed],
]
`;

exports[`createEvent should return null and an auth error without a token: not logged in 1`] = `
Array [
  [GraphQLError: You must supply a JSON Web Token for authorization, are you logged in?],
]
`;

exports[`getEvent should return an event after one has been created: event found 1`] = `
Object {
  "description": "The start of POSIX time",
  "owner": Object {
    "email": "briankernighan@example.com",
  },
  "title": "epoch",
}
`;

exports[`getEvent should return an event after one has been created: no errors 1`] = `undefined`;

exports[`getEvent should return errors for a malformed query: malformed query error 1`] = `
Array [
  [GraphQLError: Expected a valid externalId or title],
]
`;

exports[`getEvent should return errors for a query skipping a mandatory field: skipped mandatory field error 1`] = `
Array [
  [GraphQLError: Expected an event ID or title],
]
`;

exports[`getEvent should return null if no event has been found: no event error 1`] = `undefined`;

exports[`getEvent should return null if no event has been found: null event 1`] = `null`;
